common/persistence/nosql/nosqlplugin/cassandra/session.go

```diff
@@ -45,7 +44,7 @@ func CreateSession(cfg config.Cassandra) (gocql.Session, error) {
 		Datacenter:        cfg.Datacenter,
 		MaxConns:          cfg.MaxConns,
 		TLS:               cfg.TLS,
-		ProtoVersion:      cassandraProtoVersion,
+		ProtoVersion:      cfg.ProtoVersion,
```

Comment:
Can you moved the backward compatibility logic to here? We have an internal implementation for gocql and the implementation in `gocql/client.go` is not used by our internal impl. This function however, is shared by both the internal and open source impl, so having the backward compatibility logic here can ensure our internal impl won't be break.

Reply:
+1. Adding config like this should be backward compatible. Even when breaking changes is necessary, we will have a middle version for transition.

Reply:
@yycptt Done, thanks for pointing it out. Definitely trying to keep it backward compatible and let me know if you can think in another place where I should add it, I might have missed one or two places.